src/pages/gd/Claim/index.tsx

```diff
@@ -1,29 +1,123 @@
-import React, { memo } from 'react'
+import React, { memo, useMemo } from 'react'
 import { t } from '@lingui/macro'
 import { useLingui } from '@lingui/react'
-import { ClaimButton, Layout, Title } from '@gooddollar/good-design'
+import { BalanceGD, ClaimButton, ClaimCarousel, IClaimCard, Title } from '@gooddollar/good-design'
 import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { useClaim } from '@gooddollar/web3sdk-v2'
+import { Text, useMediaQuery, View } from 'native-base'
+import { useClaiming } from 'hooks/useClaiming'
+import usePromise from 'hooks/usePromise'
+import { g$Price } from '@gooddollar/web3sdk'
+import { isToday, format } from 'date-fns'
+
+const mockedCards: Array<IClaimCard> = [
+    {
+        title: 'How to claim G$',
+        content: [
+            { description: 'First time here? Watch this video to learn the basics about GoodDollar:' },
+            {
+                imageUrl:
+                    'https://1.bp.blogspot.com/-t6rZyF0sJvc/YCe0-Xx2euI/AAAAAAAADt8/ZVlJPzwtayoLezt1fKE833GRX-n8_MHWwCLcBGAsYHQ/s400-rw/Screenshot_20210213-113418.png',
+            },
+        ],
+    },
+    {
+        title: 'Claimed today? Time to use your G$. 👀',
+        content: [
+            {
+                description: `You can use your GoodDollars
+        to buy products, book services, and use DeFi to better your life and the live of others.`,
+            },
+            {
+                link: {
+                    linkText: 'Buy using G$',
+                    linkUrl: 'https://google.com',
+                },
+            },
+        ],
+    },
+    {
+        title: 'GoodDollar by numbers',
+        content: [
+            {
+                list: [
+                    { key: '🪂 Total UBI Distributed', value: '$327.5k' },
+                    { key: '💰 Unique UBI Claimers', value: '$475k' },
+                    { key: '🚢  Market Capitalization', value: '$876k' },
+                ],
+            },
+        ],
+    },
+]
 
 const Claim = memo(() => {
     const { i18n } = useLingui()
-    const { account } = useActiveWeb3React()
+    const { account, chainId } = useActiveWeb3React()
+    const { claimed, handleClaim } = useClaiming()
+    const { claimTime } = useClaim('everyBlock')
+
+    const [G$Price] = usePromise(async () => {
+        try {
+            const data = await g$Price()
+            return data.DAI
+        } catch {
+            return undefined
+        }
+    }, [chainId])
+
+    const formattedTime = useMemo(
+        () => claimed && (isToday(claimTime) ? 'today' : 'tomorrow') + ' ' + format(claimTime, 'hh a'),
+        [claimed, claimTime]
+    )
+
+    const [isSmallScreen] = useMediaQuery({
```

Comment:
1. do not rely on size, use XS-sm-md-lg instead
2. let's make reusable hook for this

```javascript
const useScreenSize = () => {
  const isLargeScreen = useBreakpointValue({ base: false, lg: true })
  const isMediumScreen = useBreakpointValue({ base: false, md: true })
  const isSmallScreen = !isLargeScreen && !isMediumScreen

  return { isSmallScreen, isLargeScreen, isMediumScreen }
}
```